### PR TITLE
close #18092 document elif in case statements

### LINKED ISSUE #18092: elif in case statement is not documented in the manual
This form is nowhere to be seen in the "Case statement" section

### Example
```nim
case true:
of false:
  doAssert false
elif true:
  echo "true"
else:
  doAssert false
```

Note that the fact that there can not be any `of` branches after an `elif` must be documented.

Found during implementation of CPS expression flattening.

### PR DESCRIPTION
**Pull request by timotheecour**
close #18092

### CODE REVIEW

**Review by alaviss — CHANGES_REQUESTED** [review]
Please also document that `elif` have to come after `of`. The manual is the language specification so we would want these details documented.

**Comment by timotheecour** [comment]
done

**Comment by Varriount** [comment]
While this technically documents "elif" in case statements (and *is* an improvement), I feel that there should be some additional explanation. Before this PR, I wasn't even aware that `case` supported `elif`, and I'm still not sure what it does, and when it should be used.

From the example, appears that `elif` allows the introduction of an arbitrary condition in a case statement? The example implies that it's only allowed at the end of a case statement, but is that true?

**Comment by timotheecour** [comment]
i feel it's clear enough and un-ambiguous as written, but feel free to suggest an alternative wording as code review comment.
>  I'm still not sure what it does, and when it should be used.

exactly what you wrote: `From the example, appears that elif allows the introduction of an arbitrary condition in a case statement?`

> The example implies that it's only allowed at the end of a case statement, but is that true?

within a case statement, yes

**Comment by alaviss** [comment]
The code example is not the spec, it's... an example.

We already have a decent amount of writing right below it, and I think we should expand the paragraph talking about `else` to talk about `elif` as well.

**Comment by timotheecour** [comment]
> The code example is not the spec, it's... an example.

yes and no; in this case I agree (and have updated spec description), but in other cases (in particular description of edge cases in APIs) it may makes sense to treat examples as spec because the alternative would be both non-DRY and add no value for readers compared to a well written (and compiler checked) runnableExamples.

I also mentioned the special case of string and mentioned the spec might change (refs https://github.com/nim-lang/Nim/issues/11474#issuecomment-504951492)

**Comment by Varriount** [comment]
The behavior an example describes tends to be implicit - my previous comment on elif's behavior was a guess. If not repeating information is a priority here, it would be preferable to go with an explicit description of elif's behavior over an implicit example.

File: doc/manual.rst [hunk 1]
```diff
@@ -2985,7 +2985,8 @@ Example:
   of "delete-everything", "restart-computer":
     echo "permission denied"
   of "go-for-a-walk":     echo "please yourself"
-  else:                   echo "unknown command"
+  elif defined(windows):  echo "bsod" # optional `elif` must come after `of` branches
```
Comment (Araq) [inline doc/manual.rst]: This example is poorly chosen as `defined` is usually used within a `when` statement.

**Comment by Araq** [comment]
The spec should say something like: "elif" in "case" is handled just like "else: if".

File: doc/manual.rst (hunk 1 above)
Comment (timotheecour) [reply doc/manual.rst]: changed to something else

**Comment by timotheecour** [comment]
PTAL

**Pull request merged**